source/backend/api/Services/DispositionFileService.cs

```diff
@@ -97,16 +100,31 @@ public PimsDispositionFile Update(long id, PimsDispositionFile dispositionFile,
                 throw new BadRequestException("Invalid dispositionFileId.");
             }
 
+            DispositionStatusTypes? currentDispositionStatus = GetCurrentDispositionStatus(dispositionFile.Internal_Id);
+            if (!_dispositionStatusSolver.CanEditDetails(currentDispositionStatus) && !_user.HasPermission(Permissions.SystemAdmin))
+            {
+                throw new BusinessRuleViolationException("The file you are editing is not active or draft, so you cannot save changes. Refresh your browser to see file state.");
+            }
+
             ValidateStaff(dispositionFile);
             ValidateVersion(id, dispositionFile.ConcurrencyControlNumber);
 
-            if (!userOverrides.Contains(UserOverrideCode.DispositionFileFinalStatus))
+            var doNotAddToStatuses = new List<string>() { EnumDispositionFileStatusTypeCode.COMPLETE.ToString(), EnumDispositionFileStatusTypeCode.ARCHIVED.ToString() };
+            var currentDispositionFile = _dispositionFileRepository.GetById(id);
+
+            if (dispositionFile?.PimsDispositionSales?.FirstOrDefault()?.SaleFinalAmt == null)
             {
-                var doNotAddToStatuses = new List<string>() { EnumDispositionFileStatusTypeCode.COMPLETE.ToString(), EnumDispositionFileStatusTypeCode.ARCHIVED.ToString() };
-                if(doNotAddToStatuses.Contains(dispositionFile.DispositionFileStatusTypeCode))
-                {
-                    throw new UserOverrideException(UserOverrideCode.DispositionFileFinalStatus, "You are changing this file to a non-editable state. Only system administrators can edit the file when set to Archived, Cancelled or Completed state). Do you wish to continue?");
-                }
+                throw new BusinessRuleViolationException("You have not added a Sales Price. Please add a Sales Price before completion.");
+            }
+            else if (currentDispositionFile.DispositionFileStatusTypeCode != EnumDispositionFileStatusTypeCode.COMPLETE.ToString()
+                && dispositionFile.DispositionFileStatusTypeCode == EnumDispositionFileStatusTypeCode.COMPLETE.ToString()
+                && currentDispositionFile.PimsDispositionFileProperties.Count > 0)
```

Comment:
this looks like good logic to be put in a different solver

Reply:
To be honest I'd rather we discuss the "solver" pattern. I think that pattern was added to the code base without a group discussion.

Reply:
Wouldn't have been better creating extension methods for the Disposition File instead of creating the "solver" approach to determine the Disposition file operability?